pyproject.toml

```diff
@@ -1,32 +1,51 @@
 [build-system]
-requires = ['flit_core >=2,<3']
-build-backend = 'flit_core.buildapi'
-
-[tool.flit.metadata]
-module = 'build'
-author = 'Filipe Laíns'
-author-email = 'lains@archlinux.org'
-home-page = 'https://github.com/FFY00/python-build'
-description-file = 'README.md'
-requires = [
-    'toml',
-    'pep517',
-    'packaging',
-    'importlib_metadata;python_version<"3.8"',
-    'typing;python_version<"3"',
+requires = ['poetry>=0.12']
```

Comment:
Why the switch to `poetry`? Seems like overkill for a super minimal tool.

Also, it seems weird that this package has both a `setup.py` and is built with something other than `setuptools`.

Reply:
> Why the switch to `poetry`? Seems like overkill for a super minimal tool.

I was using flit but it does not support specifying multiple packages.

> Also, it seems weird that this package has both a `setup.py` and is built with something other than `setuptools`.

Well, I liked the nicer CLI but I guess I can just use `setuptools`, it's not a big deal. I kept `setuptools` to be easier for packagers.

Reply:
Do you actually want to distribute multiple packages? I generally see that as an anti-pattern except in very rare situations. It's usually better to namespace them under a top-level package.

I don't think you have to change whatever you don't want to change, but I would expect that you wouldn't use `setuptools`' CLI at all, correct? Just `nox` or something of that nature (or possibly `casei` itself, for the build).

In any case, the main problem I see is that you are trying to maintain compatibility with bootstrapping via `setuptools` (or presumably `distutils`, I guess, though `distutils` will soon be part of `setuptools` and not in the standard library), so you end up needing to duplicate all the project metadata, which can easily get out of sync.

Reply:
> Do you actually want to distribute multiple packages? I generally see that as an anti-pattern except in very rare situations. It's usually better to namespace them under a top-level package.

Yes. I would really like to be able to ship a `build` module, so that plain builds (no virtual envs) can be done by invoking `python -m build`.

> I don't think you have to change whatever you don't want to change, but I would expect that you wouldn't use `setuptools`' CLI at all, correct? Just `nox` or something of that nature (or possibly `casei` itself, for the build).

Yeah, generally I wouldn't really be using the `setuptools` CLI as the package can build itself via PEP517. I just remembered my main motivation against `setuptools`, it does not create `py2.py3` wheels, only `py2` or `py3`. Either way, this is a nitpick.

> In any case, the main problem I see is that you are trying to maintain compatibility with bootstrapping via `setuptools` (or presumably `distutils`, I guess, though `distutils` will soon be part of `setuptools` and not in the standard library), so you end up needing to duplicate all the project metadata, which can easily get out of sync.

Yes, but since right now I am the only maintainer, I have it handled. That could change in the future.

Reply:
> I would really like to be able to ship a `build` module

But why is it so important to use a very generic name here?
To give the impression that this is somehow built-in to Python?

Reply:
> But why is it so important to use a very generic name here?
> To give the impression that this is somehow built-in to Python?

To give the impression it just *builds*.